packages/swingset-liveslots/src/cache.js

```diff
@@ -17,7 +17,7 @@ import { Fail } from '@agoric/assert';
 /**
  * @callback CacheDelete
  * @param {string} key
- * @returns {void}
+ * @returns {boolean}
```

Comment:
This won't work as expected (assuming that `Map`-like behavior is expected). The cache doesn't know whether a given key is present in the backing store or not, it only knows whether it was marked for deletion by a previous call to `delete()`. So the answer will depend upon the history of the cache as well as the backing store:

| present in backing store | cache operations | expected result | actual result |
| --- | --- | --- | --- |
| yes | `delete()` | true | false |
| no  | `delete()` | false | false |
| yes | `get() + delete()` | true | true |
| no | `get() + delete()` | false | true |

Liveslots doesn't have a need for a `delete()` return value, and it would cost an extra DB query to make this act like `Map`, so I'd suggest reverting this one piece.

The other change (`get()` can return `undefined` but the typedef didn't reflect that) is absolutely correct.

PR #9509 is for reverting this part.. my apologies for not reviewing this in a timely fashion.